ports/stm32f4/common-hal/busio/SPI.c

```diff
@@ -60,13 +59,39 @@ STATIC uint32_t get_busclock(SPI_TypeDef * instance) {
     return HAL_RCC_GetPCLK2Freq();
 }
 
+STATIC uint32_t stm32_baud_to_spi_div(uint32_t baudrate, uint16_t * prescaler, uint32_t busclock) {
+    static const uint32_t baud_map[8][2] = {
+        {2,SPI_BAUDRATEPRESCALER_2},
+        {4,SPI_BAUDRATEPRESCALER_4},
+        {8,SPI_BAUDRATEPRESCALER_8},
+        {16,SPI_BAUDRATEPRESCALER_16},
+        {32,SPI_BAUDRATEPRESCALER_32},
+        {64,SPI_BAUDRATEPRESCALER_64},
+        {128,SPI_BAUDRATEPRESCALER_128},
+        {256,SPI_BAUDRATEPRESCALER_256}
+    };
+    size_t i = 0;
+    uint16_t divisor;
+    do {
+        divisor = baud_map[i][0];
+        if (baudrate >= (busclock/divisor)) {
+            *prescaler = divisor;
+            return baud_map[i][1];
+        }
+        i++;
+    } while (divisor != 256);
+    //only gets here if requested baud is lower than minimum
+    *prescaler = 256;
+    return SPI_BAUDRATEPRESCALER_256;
+}
+
 void spi_reset(void) {
     uint16_t never_reset_mask = 0x00;
-    for(int i=0;i<MAX_SPI;i++) {
+    for (int i = 0; i < MAX_SPI; i++) {
```

Comment:
Not a big deal but I usually use `size_t` for unsigned indices.

Reply:
Seems like Scott uses just raw int as I do. Let's revisit this if we want to make it consistent across ports.